src/main/java/life/qbic/omero/BasicOMEROClient.java

```diff
@@ -3,384 +3,526 @@
-        this.datasetMap = new HashMap<Long, Set<DatasetData>>();
+  private SecurityContext getContext() {
+    return ctx;
+  }
 
-        try {
+  public String getSessionId() {
 
-            BrowseFacility browse = gateway.getFacility(BrowseFacility.class);
-            Collection<ProjectData> projects = browse.getProjects(ctx);
+    return this.sessionId;
+  }
 
-            Iterator<ProjectData> i = projects.iterator();
-            ProjectData project;
-            while (i.hasNext()) {
-                project = i.next();
+  public Map<Long, String> getProjectMap() {
+    return projectMap;
+  }
 
-                String name = project.getName();
-                long id = project.getId();
+  public Map<Long, Set<DatasetData>> getDatasetMap() {
+    return datasetMap;
+  }
 
-                this.projectMap.put(id, name);
-                this.datasetMap.put(id, project.getDatasets());
-            }
+  public void disconnect() {
+    this.gateway.disconnect();
+  }
 
-        } catch (Exception e) {
-            System.out.println(e);
-        }
+  public HashMap<Long, String> loadProjects() {
 
-        return this.projectMap;
-    }
+    this.projectMap = new HashMap<Long, String>();
+    this.datasetMap = new HashMap<Long, Set<DatasetData>>();
+
+    try {
 
-    public HashMap<String, String> getProjectInfo(long projectId){
+      BrowseFacility browse = gateway.getFacility(BrowseFacility.class);
+      Collection<ProjectData> projects = browse.getProjects(ctx);
 
-        HashMap<String, String> projectInfo = new HashMap<String, String>();
+      Iterator<ProjectData> i = projects.iterator();
+      ProjectData project;
+      while (i.hasNext()) {
+        project = i.next();
+
+        String name = project.getName();
+        long id = project.getId();
+
+        this.projectMap.put(id, name);
+        this.datasetMap.put(id, project.getDatasets());
+      }
+
+    } catch (Exception e) {
+      System.out.println(e);
+    }
 
-        try {
+    return this.projectMap;
+  }
 
-            BrowseFacility browse = gateway.getFacility(BrowseFacility.class);
-            Collection<ProjectData> projects = browse.getProjects(ctx);
+  public HashMap<String, String> getProjectInfo(long projectId) {
 
-            Iterator<ProjectData> i = projects.iterator();
-            ProjectData project;
-            while (i.hasNext()) {
-                project = i.next();
+    HashMap<String, String> projectInfo = new HashMap<String, String>();
 
-                if (project.getId() == projectId){
+    try {
 
-                    projectInfo.put("name", project.getName());
-                    projectInfo.put("desc", project.getDescription());
+      BrowseFacility browse = gateway.getFacility(BrowseFacility.class);
+      Collection<ProjectData> projects = browse.getProjects(ctx);
 
-                    break;
-                }
+      Iterator<ProjectData> i = projects.iterator();
+      ProjectData project;
+      while (i.hasNext()) {
```

Comment:
```suggestion
      projects.forEach(project -> {
```

Reply:
alternatively:
```suggestion
      projects.stream().filter(projectInfo -> projectInfo.getId() == projectId).forEach(projectInfo -> {
            projectInfo.put("name", project.getName());
            projectInfo.put("desc", project.getDescription());
      }
```
This would be the most concise one.